Algorithms/Search/JumpSearcher.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+namespace Algorithms.Search {
+
+    /// <summary>
+    /// The idea: This is one of the public search algorithm.
+    /// It takes 2 argument array of numbers and value of number
+    /// It finds index of this value in given array
+    ///     Time compl is O(root(n))
+    /// Note: the array has to be sorted beforehand.
+    /// </summary>
+    public class JumpSearcher {
+        /// <summary>
+        /// Code is really easy we find step size first
+        /// Later we find the block where this element should be.!
+        /// In this block we go search for value.! If found return index
+        /// Otherwise return -1!
+        /// </summary>
+        private int JumpSearch (int[] arr, int val) {
+            double temp = Math.Sqrt (val);
+            int step = (int) Math.Floor (temp);
+
+            int old = 0;
+            while (val > arr[Math.Min (step, arr.Length) - 1]) {
+                old = step;
+                temp = Math.Sqrt (arr.Length);
+                step += (int) Math.Floor (temp);
```

Comment:
Modifying step is misleading and this variable isn't named correctly. The step should be fixed and the current position should be updated every iteration. Could you refactor this?